packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -52,10 +51,45 @@ const toJsonBKey = <T extends keyof NestedFilterQueryFields>(nestedField: T): Js
   return [field, value].join(JSONB_FIELD_DELIMITER) as JsonBKey<T>;
 };
 
+// Inspired by Google Analytics filter: https://developers.google.com/analytics/devguides/reporting/core/v3/reference?hl=en#filters
+const operatorToSqlComparator = {
+  '==': null, // Exact match
+  '!=': '!=', // Does not match
+  '=@': 'ilike', // Contains sub string
+};
+const filterOperators = Object.keys(operatorToSqlComparator);
+
+const formatComparisonValue = (value: string | string[], operator: string) => {
+  if (operator === '=@' && typeof value === 'string') {
+    return `%${value}%`;
+  }
+
+  return value;
+};
+
+const convertValueToAdvancedCriteria = (operator: string, value: string | string[]) => {
+  const comparator = operatorToSqlComparator[operator];
+
+  return {
+    comparator,
+    comparisonValue: formatComparisonValue(value, operator),
+  };
+};
+
 const toFilterClause = (queryClause: string) => {
-  const clauseParts = queryClause.split(FIELD_VALUE_DELIMITER);
+  const operator = filterOperators.find(o => queryClause.includes(o));
```

Comment:
Once we cast `filterOperators` to `Operator[]` then the type of `operator` here becomes `Operator | undefined`. Doing the `if(!operator)` check later on then ensures that the type is `Operator` for the rest of the codeflow 👍